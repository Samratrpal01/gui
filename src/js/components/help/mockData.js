export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
