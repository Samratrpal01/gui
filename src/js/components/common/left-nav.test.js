import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import LeftNav from './left-nav';

describe('LeftNav Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <LeftNav
        sections={[
          {
            itemClass: 'test-list',
            items: [
              { exact: true, path: '/far/away', secondaryAction: <div>HEYA!</div>, style: { background: 'red' }, title: 'test1' },
              { path: '/far/away', title: 'test1' }
            ],
            title: 'test'
          }
        ]}
      />
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
