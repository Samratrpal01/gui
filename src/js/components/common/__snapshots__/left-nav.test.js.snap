// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav Component renders correctly 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  line-height: 48px;
  list-style: none;
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2.navLink .MuiListItemText-root {
  color: #212121;
}

.emotion-2.navLink.active .MuiListItemText-root {
  color: rgba(0, 0, 0, 0.87);
}

.emotion-2.active {
  background: #bdbdbd;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-3.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-3.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-3.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-3.Mui-disabled {
  opacity: 0.38;
}

.emotion-4 .MuiListItemText-primary {
  font-size: 0.8rem;
}

.emotion-4 .MuiListItemText-primary:hover {
  color: #1976d2;
}

.emotion-5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-7 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<ul
  class="MuiList-root MuiList-padding leftFixed emotion-0"
>
  <li
    class="MuiListSubheader-root MuiListSubheader-gutters emotion-1"
  >
    test
  </li>
  <a
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink test-list emotion-2 emotion-3"
    href="/far/away"
    style="background: red;"
  >
    <div
      class="MuiListItemText-root emotion-4 emotion-5"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-6"
      >
        test1
      </span>
    </div>
    <div
      class="MuiListItemIcon-root emotion-7"
    >
      <div>
        HEYA!
      </div>
    </div>
  </a>
  <a
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink test-list emotion-2 emotion-3"
    href="/far/away"
  >
    <div
      class="MuiListItemText-root emotion-4 emotion-5"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-6"
      >
        test1
      </span>
    </div>
  </a>
</ul>
`;
